docs/data/data-grid/server-side-data/index.md

```diff
@@ -126,161 +113,143 @@ const customDataSource: DataSource = {
-   */
-  rowCount?: number;
-  /**
-   * Additional `pageInfo` to help the grid determine if there are more rows to fetch (corner-cases)
-   * `hasNextPage`: When row count is unknown/inaccurate, if `truncated` is set or rowCount is not known, data will keep loading until `hasNextPage` is `false`
-   * `truncated`: To reflect `rowCount` is inaccurate (will trigger `x-y of many` in pagination after the count of rows fetched is greater than provided `rowCount`)
-   * It could be useful with:
-   * 1. Cursor based pagination:
-   *   When rowCount is not known, grid will check for `hasNextPage` to determine
-   *   if there are more rows to fetch.
-   * 2. Inaccurate `rowCount`:
-   *   `truncated: true` will let the grid know that `rowCount` is estimated/truncated.
-   *   Thus `hasNextPage` will come into play to check more rows are available to fetch after the number becomes >= provided `rowCount`
-   */
-  pageInfo?: {
-    hasNextPage?: boolean;
-    truncated?: number;
-  };
-}
+<DataGridPro
+  columns={columns}
+  unstable_dataSource={customDataSource} // this automatically means `sortingMode="server"`, `filterMode="server"`, `paginationMode="server"`
+/>
 ```
 
-#### Updating data
+The following demo showcases this behavior.
+
+{{"demo": "ServerSideDataGrid.js", "bg": "inline"}}
 
-If provided, the method `dataSource.updateRow` will be called with the `GridRowModel` object whenever the user edits a row. This method is optional and you can skip it if you don't need to update the data on the server. It will work in a similar way as the `processRowUpdate` prop.
+:::info
+The data source demos use a utility function `useMockServer` to simulate the server-side data fetching. In a real-world scenario, you should replace this with your own server-side data fetching logic.
 
-#### Data Grid props
+Open info section of the browser console to see the requests being made and the data being fetched in response.
+:::
 
-These data grid props will work with the server-side data source:
+## Data caching
 
-- `dataSource: DataSource`: the data source object that you need to implement
-- `rows`: will be ignored, could be skipped when `dataSource` is provided
-- `rowCount`: will be used to identify the total number of rows in the grid, if not provided, the grid will check for the _GetRowsResponse.rowCount_ value, unless the feature being used is infinite loading where no `rowCount` is available at all.
+The Data source supports caching the data it receives from the server by default. This means that if the user navigates to a page or expands a node that has already been fetched, the grid will not call the `getRows` function again to avoid unnecessary calls to the server.
 
-Props related to grouped data (`treeData` and `rowGrouping`):
+The `SimpleServerSideCache` is the cache used by default which is a simple in-memory cache that stores the data in a plain object. It could be seen in action in the demo below.
```

Comment:
How about calling it `GridDataSourceCache` instead?

Reply:
Makes sense, since its interface is bound to the GridDataSource itself.

Reply:
I'd add a `Default` since `GridDataSourceCache` is the generic interface name that users can pass from the outside.
So the name of the cache would be `GridDataSourceDefaultCache`

Reply:
Nitpick: Rename to `GridDataSourceCacheDefault` to avoid splitting the `GridDataSourceCache`? It would also make the search by `GridDataSourceCache` easier.